### PR TITLE
Handling of NSURLErrorNetworkConnectionLost error

### PR DESCRIPTION
**Pull request by mykhailo-kuchma**
Enable the handling of NSURLErrorNetworkConnectionLost network error

Resolves: OLPEDGE-2566

Signed-off-by: Mykhailo Kuchma <ext-mykhailo.kuchma@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/src/http/ios/OLPNetworkIOS.mm [hunk 1]
```diff
@@ -98,6 +98,7 @@
     case NSURLErrorDataNotAllowed:
       return ErrorCode::OFFLINE_ERROR;
     case NSURLErrorTimedOut:
+    case NSURLErrorNetworkConnectionLost:
```
Comment (andescu) [inline olp-cpp-sdk-core/src/http/ios/OLPNetworkIOS.mm:101]: Is it not rather OFFLINE_ERROR?

Comment (mykhailo-kuchma) [reply olp-cpp-sdk-core/src/http/ios/OLPNetworkIOS.mm:101]: A client or server connection was severed in the middle of an in-progress load.

https://developer.apple.com/documentation/foundation/1508628-url_loading_system_error_codes/nsurlerrornetworkconnectionlost

Comment (andescu) [reply olp-cpp-sdk-core/src/http/ios/OLPNetworkIOS.mm:101]: Yes, but this is not a time out, rather a connection lost. Maybe add a new error code to `http::ErrorCode` enum?

Comment (mykhailo-kuchma) [reply olp-cpp-sdk-core/src/http/ios/OLPNetworkIOS.mm]: @andescu, I checked other implementation, the error code should be IO_ERROR

**Review by ystefinko — APPROVED** [review]

**Pull request merged**